features/facebookImport/src/model/entities/message-threads-group.js

```diff
@@ -25,20 +25,28 @@ export default class MessageThreadsGroup {
     }
 
     forEachMessageThread(callback) {
-        for (const messageThread of this._messagesThreads) {
+        for (const messageThread of this.messagesThreads) {
             callback(messageThread);
         }
     }
 
     addMessageThreadFromData(messageThreadData) {
         const messageThread = new MessageThread();
         messageThread.initializeFromData(messageThreadData);
-        this._messagesThreads.push(messageThread);
+        const participant = messageThread.participants.toString();
+        if (this._messagesThreads[participant]) {
+            this.addToExistingThread(participant, messageThread);
+        } else this._messagesThreads[participant] = messageThread;
     }
 
     addMessageThreadsFromData(messageThreadsData) {
         messageThreadsData.forEach((messageThreadData) =>
             this.addMessageThreadFromData(messageThreadData)
         );
     }
+
+    addToExistingThread(participant, messageThread) {
+        const existingThread = this._messagesThreads[participant];
```

Comment:
OK, this is a change in API. Which makes sense (because it makes merging easier). Admittedly this is an implementation change, and you have cleverly converted what was a public attribute into a function, with the client being none the wiser.
I guess this passes tests, and everything, and you have probably considered all the implications, so I'm good with that.